File: frontend/src/lib/api/operations.ts
"use client";

import { apiRequest } from "@/lib/api/client";
import { withCompanyScope } from "@/lib/api/company-scope";
import type { ClientPanelData, DevPanelData, FinanceSummary, InventoryItem, OperationalOrder, OperationalOrderStatus } from "@/types/operations";

export function getClientPanel(companyId?: number) {
  return apiRequest<ClientPanelData>(withCompanyScope("/operations/client-panel", companyId));
}

export function getDevPanel() {
  return apiRequest<DevPanelData>("/operations/dev-panel");
}

export function getOperationalOrders(companyId?: number) {
  return apiRequest<OperationalOrder[]>(withCompanyScope("/operations/orders", companyId));
}

export function updateOperationalOrderStatus(orderId: number, status: OperationalOrderStatus, companyId?: number) {
  return apiRequest<OperationalOrder>(withCompanyScope(`/operations/orders/${orderId}/status`, companyId), {
    method: "PATCH",
    body: JSON.stringify({ status })
  });
}

export function getInventory(companyId?: number) {
  return apiRequest<InventoryItem[]>(withCompanyScope("/operations/inventory", companyId));
}

export function getFinanceSummary(companyId?: number) {
  return apiRequest<FinanceSummary>(withCompanyScope("/operations/finance", companyId));
}

File: app/api/routes/v1/operations.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.core.dependencies import require_dev, require_roles, resolve_company_id
from app.db.session import get_db
from app.schemas.operations import (
    ClientDashboardResponse,
    FinanceSummaryResponse,
    InventoryItemResponse,
    OrderResponse,
    OrderStatusUpdate,
    DevDashboardResponse,
)
from app.services.operations_service import OperationsService

router = APIRouter(prefix="/operations", tags=["operations"])


@router.get("/client-panel", response_model=ClientDashboardResponse)
def get_client_panel(
    company_id: int = Depends(resolve_company_id),
    _: object = Depends(require_roles("dev", "client", "attendant")),
    db: Session = Depends(get_db),
) -> ClientDashboardResponse:
    return OperationsService(db).get_client_dashboard(company_id)


@router.get("/dev-panel", response_model=DevDashboardResponse)
def get_dev_panel(
    _: object = Depends(require_dev),
    db: Session = Depends(get_db),
) -> DevDashboardResponse:
    return OperationsService(db).get_dev_dashboard()


@router.get("/orders", response_model=list[OrderResponse])
def list_orders(
    company_id: int = Depends(resolve_company_id),
    _: object = Depends(require_roles("dev", "client", "attendant")),
    db: Session = Depends(get_db),
) -> list[OrderResponse]:
    items = OperationsService(db).list_orders(company_id)
    return [OrderResponse.model_validate(item) for item in items]


@router.patch("/orders/{order_id}/status", response_model=OrderResponse)
def update_order_status(
    order_id: int,
    payload: OrderStatusUpdate,
    company_id: int = Depends(resolve_company_id),
    _: object = Depends(require_roles("dev", "client", "attendant")),
    db: Session = Depends(get_db),
) -> OrderResponse:
    item = OperationsService(db).update_order_status(company_id, order_id, payload.status)
    return OrderResponse.model_validate(item)


@router.get("/inventory", response_model=list[InventoryItemResponse])
def list_inventory(
    company_id: int = Depends(resolve_company_id),
    _: object = Depends(require_roles("dev", "client", "attendant")),
    db: Session = Depends(get_db),
) -> list[InventoryItemResponse]:
    items = OperationsService(db).list_inventory(company_id)
    return [InventoryItemResponse.model_validate(item) for item in items]


@router.get("/finance", response_model=FinanceSummaryResponse)
def get_finance(
    company_id: int = Depends(resolve_company_id),
    _: object = Depends(require_roles("dev", "client", "attendant")),
    db: Session = Depends(get_db),
) -> FinanceSummaryResponse:
    return OperationsService(db).get_finance_summary(company_id)
